lib/spack/docs/environments.rst

```diff
@@ -433,6 +433,18 @@ The supplied location will become the build-directory for that package in all fu
     developers to only redirect the build directory if they understand their package's
     build-system.
 
+
+Spack attempts to minimize rebuilds by checking the source directory for the last time a file
+was changed and comparing it to the installation time.
+On some filesystems, such as lustre or a NFS, this check can be quite expensive.
+Users can set the configuration option ``config:dev_specs_always_rebuild:true`` to
+disable this check.
+``config:dev_specs_always_rebuild:true`` makes it so all develop specs will perform 
+**incremental rebuilds** each time ``spack install`` is called.  Please note these are not *complete rebuilds* and will still be much quicker than a fresh installation.
+
+Adding the ``--always-rebuild`` flag to the ``spack develop`` command will automatically
```

Comment:
If we're tagging this per-package on the `spack develop` command, then we should be recording this separately for each package instead of globally. That would also allow you to get the performance benefits from packages that do incremental builds but still pay the cost to avoid full rebuilds for some packages that have long and non-incremental builds.